lib/src/rules/prefer_final_fields.dart

```diff
@@ -135,21 +152,44 @@ class _Visitor extends SimpleAstVisitor<void> {
       return;
     }
 
-    fields.variables.forEach((VariableDeclaration variable) {
+    for (var variable in fields.variables) {
       final element = variable.declaredElement;
-      if (!element.isPrivate) {
-        return;
-      }
-
-      if (variable.initializer == null) {
-        return;
-      }
 
-      if (_mutatedFields.contains(element)) {
-        return;
+      if (element.isPrivate && !_mutatedFields.contains(element)) {
+        bool fieldInConstructor(constructor) =>
+            constructor.initializers.any((initializer) =>
+                _containedInInitializer(element, initializer)) ||
+            constructor.parameters.parameters
+                .any((formal) => _containedInFormal(element, formal));
+
+        final classDeclaration = node.parent;
+        final constructors = (classDeclaration is ClassDeclaration
+            ? classDeclaration.members
+                .whereType<ConstructorDeclaration>()
+                .toSet()
```

Comment:
I don't think there's any advantage to using a `Set` rather than a `List` (or even an `Iterable`), and there might be a small performance penalty for doing so.